packages/cli/src/util/link/repo.ts

```diff
@@ -0,0 +1,269 @@
+import chalk from 'chalk';
+import pluralize from 'pluralize';
+import { homedir } from 'os';
+import { join, normalize } from 'path';
+import { normalizePath } from '@vercel/build-utils';
+import { lstat, readJSON, outputJSON } from 'fs-extra';
+import confirm from '../input/confirm';
+import toHumanPath from '../humanize-path';
+import { VERCEL_DIR, VERCEL_DIR_REPO } from '../projects/link';
+import { getRemoteUrls } from '../create-git-meta';
+import link from '../output/link';
+import { emoji, prependEmoji } from '../emoji';
+import selectOrg from '../input/select-org';
+import type Client from '../client';
+import type { Project } from '@vercel-internals/types';
+
+const home = homedir();
+
+export interface RepoProjectConfig {
+  id: string;
+  name: string;
+  directory: string;
+}
+
+export interface RepoProjectsConfig {
+  orgId: string;
+  remoteName: string;
+  projects: RepoProjectConfig[];
+}
+
+export interface RepoLink {
+  rootPath?: string;
+  repoConfigPath?: string;
+  repoConfig?: RepoProjectsConfig;
+}
+
+/**
+ * Given a directory path `cwd`, finds the root of the Git repository
+ * and returns the parsed `.vercel/repo.json` file if the repository
+ * has already been linked.
+ */
+export async function getRepoLink(cwd: string): Promise<RepoLink> {
+  let repoConfigPath: string | undefined;
+  let repoConfig: RepoProjectsConfig | undefined;
+
+  // Determine where the root of the repo is
+  const rootPath = await findRepoRoot(cwd);
+
+  // Read the `repo.json`, if this repo has already been linked
+  if (rootPath) {
+    repoConfigPath = join(rootPath, VERCEL_DIR, VERCEL_DIR_REPO);
+    repoConfig = await readJSON(repoConfigPath).catch(err => {
+      if (err.code !== 'ENOENT') throw err;
+    });
+  }
+
+  return { rootPath, repoConfig, repoConfigPath };
+}
+
+export async function ensureRepoLink(
+  client: Client,
+  cwd: string,
+  yes = false
+): Promise<RepoLink | undefined> {
+  const { output } = client;
+
+  let { rootPath, repoConfig, repoConfigPath } = await getRepoLink(cwd);
+  if (rootPath) {
+    output.debug(`Found Git repository root directory: ${rootPath}`);
+  } else {
+    throw new Error('Could not determine Git repository root directory');
+  }
+
+  // If `repoPath` is defined then this should always be defined
+  if (!repoConfigPath) {
+    throw new Error('`repoConfigPath` not defined');
+  }
+
+  if (!repoConfig) {
+    // Not yet linked, so prompt user to begin linking
+    let shouldLink =
+      yes ||
+      (await confirm(
+        client,
+        `Link to Git repository at ${chalk.cyan(
+          `“${toHumanPath(rootPath)}”`
+        )}?`,
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏make it clear here too that this supports multiple projects

```suggestion
        )} to your Project(s)?`,
```